Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using Microsoft.MixedReality.Input;
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input
+{
+    public class MotionControllerState
```

Comment:
Can this be `internal`? Maybe `protected internal`, if we expect derived classes to need to know about it?